R/to_xml.R

```diff
@@ -87,6 +87,22 @@ transform_block <- function(code_block){
   # This prevents partial code blocks that are still apparently valid: ```{r, }
   info <- info[names(info) != ""]
 
+  outchunk_option <- !(names(info) %in% c("language", "name"))
+  names(info)[outchunk_option] <- paste0(names(info)[outchunk_option], "-outchunk")
+
+  code <- strsplit(xml2::xml_text(code_block), "\n")[[1]]
+  inchunk_info <- knitr::partition_chunk(info[["language"]], code)
+  xml2::xml_text(code_block) <- paste0(inchunk_info$code, "\n")
```

Comment:
Also, I was wondering why the Julia example was being truncated and it's because of this nasty property of the text for code chunks: the input needs to be a single length vector

```suggestion
  xml2::xml_text(code_block) <- paste0(paste(inchunk_info$code, collapse = "\n"), "\n")
```

Reply:
Oh, good catch!!